pysages/backends/hoomd.py

```diff
@@ -197,20 +197,20 @@ def bind(
     wrapped_context: ContextWrapper, sampling_method: SamplingMethod, callback: Callable, **kwargs
 ):
     context = wrapped_context.context
+    sysview = SystemView(context.system_definition)
+    wrapped_context.view = sysview
+    wrapped_context.run = hoomd.run
     helpers, restore, bias = build_helpers(context, sampling_method)
 
-    with SystemView(context.system_definition) as sysview:
-        wrapped_context.view = sysview
-        wrapped_context.run = hoomd.run
-
+    with sysview:
         snapshot = take_snapshot(wrapped_context)
```

Comment:
So, the only function we call inside the context manager is the snapshot.
But don't we lose the context manager access for the snapshot once we leave the sysview context?

Seems to me like we use a dangling pointer here.

Reply:
We won't because, we would be returning copies of all arrays.